src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs

```diff
@@ -419,11 +419,22 @@ public void TestPreviewFeatures(bool enablePreviewFeatures, bool generateRequire
             var values = getValuesCommand.GetValues();
             var langVersion = values.FirstOrDefault() ?? string.Empty;
 
-            if (generateRequiresPreviewFeaturesAttribute)
+            if (enablePreviewFeatures && generateRequiresPreviewFeaturesAttribute)
             {
-                Assert.True(contains);
+                if (targetFramework == "net6.0")
```

Comment:
This test is going to need to be updated every time we bump the major version.  Possible ways to avoid this:

- Have a way for a test to get the "current" version of .NET.  We've had a bit of prior discussion on this, it would help us to keep our test coverage more current.
- Run `dotnet new console` to and read the TargetFramework from the created project.  This should be the latest .NET version (once the templates are updated after we start work on a new version).

Reply:
> Run dotnet new console to and read the TargetFramework from the created project.

I ended up going down this path and hacked up a static getter. I _think_ it'll work across all our environments. Open to feedback here!